### PR TITLE
Add some "could be written as" examples

### LINKED ISSUE #4405: Missing "could be written as"
There are some lints missing a "could be written as".
If I am on a clippy page, like [manual_memcpy](https://rust-lang.github.io/rust-clippy/master/index.html#manual_memcpy), I really like to not just see what is wrong in the example but also see how to fix the example.

Is there an interest in adding some more "could be written as" like in #4404 ?

I would be happy to fill some PRs for this :)

### PR DESCRIPTION
**Pull request by BO41**
fixes #4405
changelog: none

### CODE REVIEW

**Comment by phansch** [comment]
Thanks, the build will currently fail due to #4432 so don't worry too much about it =)

File: clippy_lints/src/methods/mod.rs [hunk 1]
```diff
@@ -425,6 +430,11 @@ declare_clippy_lint! {
     /// # let vec = vec![1];
     /// vec.iter().find(|x| **x == 0).is_some();
     /// ```
+    /// Could be written as
+    /// ```rust
+    /// # let vec = vec![1];
+    /// vec.iter().any(|x| **x == 0);
```
Comment (phansch) [inline clippy_lints/src/methods/mod.rs]: This fails on travis with

```
error[E0614]: type `{integer}` cannot be dereferenced
 --> doctest.rs:27:20
  |
4 | vec.iter().any(|x| **x == 0);
  |                    ^^^
```

You should be able to run it locally with `cargo test --doc SEARCH_IS_SOME`

Comment (BO41) [reply clippy_lints/src/methods/mod.rs]: but then the lint shouldn't make this a recommendation :confused: 
>= note: replace `find(|x| **x == 0).is_some()` with `any(|x| **x == 0)`

Comment (flip1995) [reply clippy_lints/src/methods/mod.rs]: cc #4453

**Pull request merged**